app/src/main/java/is/xyz/mpv/MPVActivity.kt

```diff
@@ -506,6 +527,64 @@ class MPVActivity : Activity(), MPVLib.EventObserver, TouchGesturesObserver {
         return true
     }
 
+    private fun interceptDpad(ev: KeyEvent): Boolean {
+        if (btnSelected == -1) { // UP and DOWN are always grabbed and overriden
+            if (ev.keyCode == KeyEvent.KEYCODE_DPAD_UP || ev.keyCode == KeyEvent.KEYCODE_DPAD_DOWN) {
+                if (ev.action == KeyEvent.ACTION_DOWN) { // activate dpad navigation
+                    btnSelected = 0
+                    updateShowBtnSelected()
+                }
+                return true
+            }
+        } else { // only used when dpad navigation is active
+            if (ev.keyCode == KeyEvent.KEYCODE_DPAD_UP || ev.keyCode == KeyEvent.KEYCODE_DPAD_DOWN) {
+                if (ev.action == KeyEvent.ACTION_DOWN) { // deactivate dpad navigation
+                    btnSelected = -1
+                    updateShowBtnSelected()
+                }
+                return true
+            } else if (ev.keyCode == KeyEvent.KEYCODE_DPAD_RIGHT) {
+                if (ev.action == KeyEvent.ACTION_DOWN) {
+                    val childCount = controls_button_group.getChildCount();
+                    btnSelected = (btnSelected + 1) % childCount
+                    updateShowBtnSelected()
+                }
+                return true
+            } else if (ev.keyCode == KeyEvent.KEYCODE_DPAD_LEFT) {
+                if (ev.action == KeyEvent.ACTION_DOWN) {
+                    val childCount = controls_button_group.getChildCount();
+                    btnSelected = (childCount + btnSelected - 1) % childCount
+                    updateShowBtnSelected()
+                }
+                return true
+            } else if (ev.keyCode == KeyEvent.KEYCODE_ENTER || ev.keyCode == KeyEvent.KEYCODE_DPAD_CENTER) {
+                if (ev.action == KeyEvent.ACTION_DOWN) {
+                    val child = controls_button_group.getChildAt(btnSelected)
+                    if (child != null) {
+                        dpadCenterTriggerAction(child.id)
+                    }
+                }
+                return true
+            }
+        }
+        return false
+    }
+
+    fun updateShowBtnSelected () {
+        if (controls_button_group != null) {
```

Comment:
`controls_button_group` will never be null

Reply:
Removed.